src/lib/ext-activitylog.js

```diff
@@ -135,9 +175,31 @@ class View {
     this.keywordFilter = document.querySelector('filter-keyword');
     this.timestampFilter = document.querySelector('filter-timestamp');
 
+    this.optionsBtn.addEventListener('click', this);
     this.clearLogBtn.addEventListener('click', this);
     this.saveLogBtn.addEventListener('click', this);
     this.loadLogFile.addEventListener('change', this);
+    this.logView.addEventListener('logcountchange', this);
+
+    this.extFilter.addEventListener('triggerdropdown', this);
+    this.viewTypeFilter.addEventListener('triggerdropdown', this);
+    this.apiTypeFilter.addEventListener('triggerdropdown', this);
+    this.apiNameFilter.addEventListener('triggerdropdown', this);
+    this.timestampFilter.addEventListener('triggerdropdown', this);
+    this.optionsBtn.addEventListener('triggerdropdown', this);
+
+    this.optionsBtn.toggleDropdown = ({ displayPref }) => {
+      switch (displayPref) {
+        case 'show':
+          this.optionsDropdown.hidden = false;
```

Comment:
The `optionsDropdown` element is a child of the `optionsBtn` element. There is no need for a custom JS method to toggle the visibility, please use CSS. This applies to almost every part in this patch that toggles the `hidden` attribute.

Reply:
I have removed the method and I have added/removed CSS class to deal with showing/hiding dropdown now .